File: frontend/src/app/models/demandeRetour.ts
import { command } from "ngx-bootstrap-icons";
import { Client } from "./client";
import { User } from "./user";
import { Commande } from "./Commande";
import { LigneDemandeRetour } from "./ligneDemandeRetour";

export class DemandeRetour{
    id:number
    nbArtcileRetenue:number
    userCreatedBy:User
    userId:string
    client:Client
    clientId:number
    dateCreated:Date
    admin:User 
    adminId:string
   
    lignes:LigneDemandeRetour[]
    commande:Commande
    commandeId:number
    constructor(Id:number,nbArtcileRetenue:number,userCreatedBy:User,userId:string,client:Client,
        clientId:number,dateCreated:Date,admin:User,adminId:string,lignes:LigneDemandeRetour[],
        commande:Commande,commandeId:number){
this.id=Id
this.nbArtcileRetenue=nbArtcileRetenue
this.userCreatedBy=userCreatedBy
this.userId=userId
this.client=client
this.clientId=clientId
this.dateCreated=dateCreated
this.admin=admin
this.adminId=adminId
this.lignes=lignes
this.commande=commande
this.commandeId=commandeId
    }
}

File: backend/B2BApplication/Models/DemandeRetour.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace B2BApplication.Models
{
    public class DemandeRetour
    {
        [Key]
    public int Id { get; set; }
        public int nbArtcileRetenue { get; set; }
        public User userCreatedBy { get; set; }
        [ForeignKey("userCreatedBy")]
        public string userId { get; set; }
        public Client client { get; set; }
        [ForeignKey("client")]
        public int clientId { get; set; }
        public DateTime dateCreated { get; set; }= DateTime.Now;
        
        public User? admin { get; set; }
        [ForeignKey("admin")]
        public string? adminId { get; set; }
       
        public ICollection<LigneDemandeRetour> lignes { get; set; }
        public Commande commande { get; set; }
        public int commandeId { get; set; }

    }
}
